datalad/support/gitrepo.py

```diff
@@ -4011,6 +4077,19 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
                                 else tuple())}):
                 yield r
 
+        # and lastly fixup subdataset states
+        for sub in [f for f, props in status.items()
```

Comment:
Oy, so I think we're looping over all tracked items in the repo again.  Can this be guard by an "on adjusted branch" condition?  Also, conceptually this seems a better fit for `AnnexRepo._save_post`.